Automattic/wp-calypso
# Redesign the all domains list component

#### Changes proposed in this Pull Request

* Add new `AllDomains` component that'll render the new all domains list redesigned view
* Fixed the ordering by Site in the all domains list
* Added filter dropdown in the all domains list
* Refactor the `DomainRow` component to remove duplicate `domainDetails` prop
* Add Site column in the `DomainRow` component when rendering the all domains view
* Fix how the `selectedSite` is passed down to the `DomainsTable` component
* In order to make this work I'm loading all the domain data initially - it might be slow but all the filtering and sorting should work

What's done and what's left that I'll fix in a follow up PRs: 
- [ ] add loading state for the filter dropdown
- [ ] add loading state for the `DomainRow` component
- [ ] fix the auto-renew toggle but (sometimes it stays in loading state forever)
- [ ] add domain ownership value in the REST API that we should use for the domain filter
- [ ] fix the bulk contact editing
- [ ] show the filter dropdown on mobile
- [x] lazy load the detailed domain data - we'll load all the domain data for now (we won't use the lazy loading 
- [x] render the site as link to the site domains list (and how to render domain-only sites)
- [x] make the sorting by site column work
- [x] render the site in mobile view
- [x] refactor the `domain`/`domainDetails` prop in the DomainRow component (we don't need both)
- [x] add the filter button

It should look like this:

![Screenshot 2021-11-12 at 14 16 28](https://user-images.githubusercontent.com/1355045/141467179-3c7591c7-ad09-4a75-ae15-3027e5d69ef3.png)
![Screenshot 2021-11-12 at 14 16 53](https://user-images.githubusercontent.com/1355045/141467182-83c9936c-955c-4eba-a7a1-f41ea3f9ca3c.png)


#### Testing instructions

* Spin up this branch or calypso.live and go to /domains/manage . Verify that the all domains list renders as expected on desktop and mobile and that the sorting works
* Some of the functionality might not work as expected (see the above list)
